crates/stages/src/pipeline/mod.rs

```diff
@@ -300,6 +300,8 @@ where
 
                         self.listeners
                             .notify(PipelineEvent::Unwound { stage_id, result: unwind_output });
+
+                        tx.commit()?;
```

Comment:
Makes sense - we move the commit from outside of the loop to inside given we want the unwind to respect the commit thresholds